src/parser/rogue/shared/resources/EnergyCapTracker.js

```diff
@@ -38,8 +38,13 @@ class EnergyCapTracker extends RegenResourceCapTracker {
   static exemptFromRefund = [
   ];
 
+  increasedBaseRegen(){
+    return 0;
+  }
+
   naturalRegenRate() {
     let regen = super.naturalRegenRate();
+    regen += this.increasedBaseRegen();
```

Comment:
Is there a better way to accomplish this? Buried Treasure is a modifier to base regen, applied before the Vigor multiplier. It didn't look like I could just modify the static base regen property as it needs to be dynamic

Reply:
I think you want to update `static baseRegenRate = BASE_ENERGY_REGEN;` based on this buff, so maybe change it to be a function. 

You can also try overriding  `naturalRegenRate()` by applying Buried Treasure or (Buried Treasure*Vigor).
But  not sure if its better then your solution.

Reply:
I gave it a shot to move the adjustment logic to be just within the subclass, but it still feels a bit odd. In my override of naturalRegenRate(), I just call an update to set baseRegenRate as the first step, before going up to super.naturalRegenRate(). 

Having baseRegenRate be a function seems like it'd be ideal, but I think that would require the RegenResourceCapTracker to be implementing it as a function, right?

Reply:
Not a JS expert, so cant tell you the optimal way to do this, but you can chanage the RegenResourceCapTracker. Clearly you have a valid use case for it. There are only 2 usages of it right now so it wont be hard to do

Reply:
Gave that a shot in the last commit. I just added a new method in the RegenResourceCapTracker which by default returns the baseRegenRate. This way it doesn't have any impact to the other specs using it and can just be overridden as required